pkg/cmd/promote.go

```diff
@@ -468,6 +469,12 @@ func (o *promoteCmdOptions) editIntegration(it *v1.Integration) *v1.Integration
 		dst.Spec.Traits.Container = &traitv1.ContainerTrait{}
 	}
 	dst.Spec.Traits.Container.Image = contImage
+	if dst.Spec.Traits.JVM == nil {
```

Comment:
I don't think we need this check. We know the promote tests are kind of flaky, no need to add anything in the command IMO. The container has to be initialized because it could raise a nil exception, but not the JVM trait, for which, we must keep the default setting.

Reply:
It is not a flacky test in this case, the promote command run the "promoted" integration with the trait `Container.Image`, so it becomes an external kit. The alternative would be to add support for the traits on the comand options.

Reply:
I see. We can proceed then.